mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -1564,6 +1565,32 @@ protected void renameMailboxByIdShouldChangeTheMailboxPathOfAMailbox() throws Ex
                 .isEqualTo(mailboxPath2);
         }
 
+        @Test
+        void renameMailboxShouldThrowWhenMailboxPathDoesNotBelongToUser() throws Exception {
+            MailboxSession sessionUser1 = mailboxManager.createSystemSession(USER_1);
+            MailboxSession sessionUser2 = mailboxManager.createSystemSession(USER_2);
+
+            MailboxPath mailboxPath1 = MailboxPath.forUser(USER_1, "mbx1");
+            MailboxPath mailboxPath2 = MailboxPath.forUser(USER_1, "mbx2");
+            mailboxManager.createMailbox(mailboxPath1, sessionUser1);
+
+            assertThatThrownBy(() -> mailboxManager.renameMailbox(mailboxPath1, mailboxPath2, sessionUser2))
+                .isInstanceOf(MailboxException.class);
+        }
+
+        @Test
+        void renameMailboxByIdShouldThrowWhenMailboxPathDoesNotBelongToUser() throws Exception {
+            MailboxSession sessionUser1 = mailboxManager.createSystemSession(USER_1);
+            MailboxSession sessionUser2 = mailboxManager.createSystemSession(USER_2);
+
+            MailboxPath mailboxPath1 = MailboxPath.forUser(USER_1, "mbx1");
+            MailboxPath mailboxPath2 = MailboxPath.forUser(USER_1, "mbx2");
```

Comment:
shouldn't it be USER_2?

Reply:
Could you add the test where USER_1 is renaming a mailboxPath to one of USER_2
and when he rename a USER_2 path to one of its own ?

Reply:
Not necessary, the fact that user 2 is trying and should not be able to rename a mailbox of user1 to an other mailbox of user1 is valid I believe. But maybe you want then an other test with your example?

Reply:
Yes I think @remk got the point I raised, but we wrote a comment on the same time. Will add those then !

Reply:
Raised an other issue to fix at the end, thanks for the feedback, good catch !